addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRuleTest.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {ACTUATOR_CONTENT_TYPE, "Content-Type: application/json"})
+    void shouldAlertIfTestedUrlRespondsOKWithRelevantContent(String contentType)
+            throws HttpMalformedHeaderException {
+        // Given
+        String servePath = "/shouldAlert";
+        String alertPath = REQ_PATH;
+        this.nano.addHandler(
+                new NanoServerHandler(servePath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, IRRELEVANT_BODY);
+                    }
+                });
+        this.nano.addHandler(
+                new NanoServerHandler("/" + alertPath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, contentType, RELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(servePath);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertEquals(httpMessagesSent.size(), 1);
+        Alert alert = alertsRaised.get(0);
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+        assertEquals(rule.getName(), alert.getName());
+        assertEquals(rule.getWascId(), alert.getWascId());
+    }
+
+    @Test
+    void shouldNotAlertIfTestedUrlRespondsOKWithIrrelevantContent()
+            throws HttpMalformedHeaderException {
+        // Given
+        String servePath = "/shouldAlert";
+        String alertPath = REQ_PATH;
+        this.nano.addHandler(
+                new NanoServerHandler(servePath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, IRRELEVANT_BODY);
+                    }
+                });
+        this.nano.addHandler(
+                new NanoServerHandler("/" + alertPath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, ACTUATOR_CONTENT_TYPE, IRRELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(servePath);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+        assertEquals(2, httpMessagesSent.size());
```

Comment:
It seems funny (likely a copy/paste) that these both assert size but in two different ways. I dunno which is best. But consistency is probably a good thing. (Through out this unittest class)